### PR TITLE
[PLA-9883] Expose predictor versioning.

### PR DESCRIPTION
**Pull request by anoto-moniz**
Adds support to the predictor collection for performing operations on specific predictor versions. Also automatically uses a new feature to cut a version upon successful predictor training.

This is split into three commits.
- One drops the patch we added to preserve the user's understanding of what archive means, since the ability to separately archive roots and versions confuses that. With the release of versions, this is no longer a concern.
- One adds support for versions to the predictor collection.
- One add support for predictor versions to collections which hit related endpoints.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (non-breaking change to assist developers)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [x] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in __version\__.py

### CODE REVIEW

**Comment by anoto-moniz** [comment]
Moved this back to a draft, as I need to add versions support to some other parts of the SDK. I'll set it back to ready and request new reviews once that's up.

File: tests/resources/test_predictor.py
Comment (genfx999) [inline tests/resources/test_predictor.py:93]: Was removing this intentional?

**Comment by anoto-moniz** [comment]
Note we're going to hold off merging until UI is ready to release, which won't be for a bit. As such, I'm leaving it in draft, so if I need to make relevant updates/changes for them, you'll still be viewing it all at once.

**Review by genfx999 — APPROVED** [review]
A question about a removed assertion, but otherwise LGTM.  Also should let Ken get a look before merging.

File: tests/resources/test_predictor.py
Comment (anoto-moniz) [reply tests/resources/test_predictor.py:93]: Yes. Currently, `archived_predictor` is just the payload we told it to respond with on line 82. Previously, the value returned by `is_archived` would be impacted by processing on the response of a subsequent call, in order to present a value the user expected before they were exposed to versions.

**Review by genfx999 — DISMISSED** [review]

**Review by genfx999 — APPROVED** [review]

**Pull request merged**
